app/utils/withPreparedDispatch.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import { withPreparedEffect } from '@webkom/react-prepare';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import type { Dispatch } from 'app/types';
+
+type PreparedEffectOptions = {
+  runSync: boolean,
+};
+
+const mapDispatch = (dispatch) => ({ dispatch });
+const mapState = () => ({});
+const injectDispatch = connect(mapState, mapDispatch);
+
+/**
+ * A higher order component that calls prepareFn
+ * whenever one of the given watchProps change.
+ *
+ * watchProps supports any strings that can be passed to _.get.
+ *
+ * opts are the options given to 'prepared()'. 'opts' will override
+ * values generated inside prepare()
+ */
+
+export default function withPreparedDispatch(
+  key: string,
+  prepareDispatchFn: (props: Object, dispatch: Dispatch<*>) => ?Promise<*>,
+  depsFn?: (props: Object) => Array<*>,
+  options?: PreparedEffectOptions
+): any {
+  return compose(
+    injectDispatch,
```

Comment:
Why is this here? All the routes should be connected already, so this will just add a bunch of overhead?

Reply:
No, they are connected after `withPreparedDispatch`. This makes sense because, in theory‚ there could be another prepared effect deeper in the component tree, that uses a prop from `connect` that is fetched in the `withPreparedDispatch`. The `connect` needs to be below `withPreparedDispatch`, so that the data will be awaited before connecting.

However we do need access to `dispatch` in order to execute the fetch action, so we need another `connect` before our `witchPreparedEffect`, only providing us with `dispatch`.